magma/circuit.py

```diff
@@ -145,6 +184,19 @@ def __repr__(cls):
             return f"{name} = DeclareCircuit(\"{name}\", {args})"
         s = f"{name} = DefineCircuit(\"{name}\", {args})\n"
         sorted_instances = sorted(cls.instances, key=lambda x: x.name)
+
+        values = []
```

Comment:
This will treat circuits with different intermediate values as unique, whereas before if they had the same structure ignoring intermediates (since ports would be directly connected), they would be treated as the same.  This shouldn't break anything, since it will be more conservative, but it may uniquify in cases where the structure is identical but intermediates used were different.  This is desired in the case of debug mode (when we are code generating the intermediates).  I would say that this would be a better default, and to "optimize" we could set a flag (or check nodebugmode) and ignore intermediates in the uniqueness check.